### ISSUE TITLE
Clear recent projects does not work in VS 2017 after the last update

### ISSUE THREAD

**Issue by FrancescaY** [Microsoft/VS-PPT#167]
Power Commands Addin did function for me until I (yesterday) downloaded the last VS 2017 upgrade (15.5.3). Now the option "Clear recent projects" does not appear at all (see image).
![untitled](https://user-images.githubusercontent.com/24878927/34939288-9958cc54-f9eb-11e7-84e2-9592351abdae.png)
I (not me alone, I believe!) need something to clear the MRU projects which is working stably and not ceasing to work after new silly VS update (which one cannot avoid).
This IS annoying!
Thanks in advance for taking your blessed measures!

**Comment by sihol** [comment]
I'm also seeing the same issue.

**Comment by justcla** [comment]
Thanks for bringing this to our attention.

Out of interest, why is it so important for you to clear the MRU projects list? I normally just let it grow. The ones I want are normally at the top.
What reason do you have for wanting the list cleared?

**Comment by FrancescaY** [comment]
I simply do not allow any VS 2017 updates to install. Too risky.

S.

**Comment by justcla** [comment]
I like updating because then I get the new features. In 15.5 we brought in Expand/Contract selection. And in 15.6 we're getting Duplicate code. :-)

**Comment by sihol** [comment]
The reason I wanted to clear it in this instance was that I had moved all my projects from one directory to another and the MRU projects list was no longer valid.

**Comment by justcla** [comment]
I think you can remove the items individually from the Start Page.

Otherwise, they will eventually go away when you open more projects.

**Comment by FrancescaY** [comment]
This is, as always, a matter of taste. I never use the Start Page, out 
of historical reasons.

**Comment by justcla** [comment]
Well, I don't know historic reasons you are referring to. But I believe it will allow you to achieve your goals, and is my recommended workaround for clearing the MRU list while the PPT feature is broken.

**Comment by Kaelum** [comment]
@justcla If you work on many different branches of the same code, it's nice to have this list reflect ONLY the Main branch, and not any of the release branches.  When I am required to go into a release branch and make changes, I immediately remove those entries from the list, as I don't want to accidentally make changes to those branches.

If you are asking, "Why is this even an issue?", well it is an issue due to the fact that we CAN'T see the full path to files/projects/solutions in the Visual Studio 2017 UI, it is clipped.  Even when you hover over them, they are still clipped, and there is no way of knowing which file it is until you physically open it and look at the file properties.

**Comment by justcla** [comment]
@Kaelum Thanks for the explanation.

> If you are asking, "Why is this even an issue?", well it is an issue due to the fact that we CAN'T see the full path to files/projects/solutions in the Visual Studio 2017 UI, it is clipped. Even when you hover over them, they are still clipped, and there is no way of knowing which file it is until you physically open it and look at the file properties.

You've brought up a major usability issue with the Start Page.
This deserves its own bug/issue on the VS IDE team.

**Comment by Kaelum** [comment]
@justcla the truncation is in the **FILE** / **Recent Projects and Solutions** list.  The hover over on the **Start Page** does appear to show the full path, but it doesn't always work.  I sometimes have to move the cursor away and back as many as 5 times before the tooltip will popup.  Since this is a browser window, the tooltip times out, which is another PITA.  Tooltips should never timeout, especially in Visual Studio.  For some reason, pretty much every browser page/window in VS doesn't always work correctly, especially if they use JavaScript, or whether is being used to make them interactive.

**Comment by mattGuima** [comment]
also if you want to screen capture to create content such as tutorials for the net you don't want to expose stuff you've been working on, especially projects under NDA! similar case when using VS for demoing different things for different clients. Anyway, clearing that list really should be a feature. long overdue.

**Comment by eman1986** [comment]
I too would like to see this fixed, I'm super OCD about having a clean list so I'm feeling dirty at the moment. I also don't use the start menu as it seems pointless to me and just clutter.

**Comment by joepeavey** [comment]
Experiencing something similar, except for me, I can remove a project in the list with the context menu. That is, it's removed, but after a short delay, the project is re-added by VS 2017. It's repeatable.

As to the guy that ask "Why do you want to remove items", who are you to try to tell anyone how they want to organize their project list? Seriously? I simply like keeping it organized, not that I have to justify to you how I use VS 2017.....

**Comment by francisrohner** [comment]
Found myself facing the same problem with VS2017.
I just migrated a bunch of projects to a new location.
If you want to cleanup your Recent Projects list:

Open regedit, and goto
HKEY_CURRENT_USER\Software\Microsoft\VisualStudio\11.0\ProjectMRUList

Here you will see a bunch of keys File1, File2, ...
Delete the items you no longer wish to see.

P.S. - Not entirely sure if you have to renumber them.

**Comment by GammaFour** [comment]
I agree with JackReacher.  Further, VS2017 will add projects to the MRU list when you're just looking at them.  So if I pull down a solution from GitHub and browse around in it, when try and open up VS2017 again I see the list polluted with all the .CSPROJ files from my browsing.  I like to keep the MRU tidy with just my actual, recently used items.

**Comment by GammaFour** [comment]
I don't know if this is the same issue, but I have a project, PROJECT1.CSPROJ that won't go away no matter how many times I delete it from the MRU list.

**Comment by Kaelum** [comment]
For those who haven’t figured it out, this list in now controlled by the Startup page.  Simply right click and remove items from this page, and they are gone.  The page sometimes gets into a weird state where it stops removing items. When this happens, close Visual Studio, reopen it, and continue on.

**Comment by francisrohner** [comment]
> For those who haven’t figured it out, this list in now controlled by the Startup page. Simply right click and remove items from this page, and they are gone. The page sometimes gets into a weird state where it stops removing items. When this happens, close Visual Studio, reopen it, and continue on.

That works very well as long as you're not removing 15+ projects.

**Comment by francisrohner** [comment]
Hmmm the registry key that I gave previously seems to be irrelevant to VS2017.
I did some searching around the registry and didn't seem to find anything for VS2017.
However I found something interesting in a file under AppData:
%USERPROFILE%\AppData\Local\Microsoft\VisualStudio\15.0_2caad97f\ApplicationPrivateSettings.xml

This XML document contains a node with the attribute name and value "CodeContainers.Offline".
The content of the node is JSON containing projects and last accessed date/time values.

I'm guessing cleaning up the JSON contents in this file would result in a clean Recent Projects list.
So far I've had no luck experimenting with this though.
After closing VS2017, Notepad++ is still telling me a process has the file open.

**Comment by Phen-Ro** [comment]
Over a year and a half later, I'm here to report that this bug is still present in VS 16.8.  And that ApplicationPrivateSettings.xml is still an XML file full of JSON.

The fix of manually clearing out the contents of CodeContainers.Offline node in ApplicationPrivateSettings.xml still works.

**Comment by YordanYanakiev** [comment]
I have multiple version of VS, and the list stuck with the VS 2017, and since then nothing reflects the changes in the newer VS's, but they all show the stuck state in 2017.

**Comment by PaulCoddington** [comment]
> @Kaelum Thanks for the explanation.
> 
> > If you are asking, "Why is this even an issue?", well it is an issue due to the fact that we CAN'T see the full path to files/projects/solutions in the Visual Studio 2017 UI, it is clipped. Even when you hover over them, they are still clipped, and there is no way of knowing which file it is until you physically open it and look at the file properties.
> 
> You've brought up a major usability issue with the Start Page. This deserves its own bug/issue on the VS IDE team. Thanks for bringing it to my attention.
> 
> Edit: I've just sent internal email about the Start Page path truncation issue.

Does Visual Studio 2022 store MRU somewhere non-cryptic and accessible?

Previous versions of Visual Studio had simple MRU keys in the Registry which were easily added to a Clear MRU script that clears all MRUs for all applications in one hit, making life much easier when swapping tasks/projects, given MRUs quickly go out of date and end up as frustrating visual clutter.

There seems to be no interface to clear MRU in the main IDE File menu, not even by temporarily setting the count to zero (not allowed). So, it ends up with dead links to files that no longer exist and records files that were accidentally opened from Explorer by double-clicking instead 'Open with'.

Another downside of not allowing users to clear MRU is that the "solution" to this problem that turns up in searches (and is also suggested by Bing Copilot) is to run '.\devenv.exe /resetuserdata', which does not merely clear the MRU but clears all settings (not good).

**Comment by ghost** [comment]
@PaulCoddington the easiest way to clear the MRU list is via the `Start Window` popup.  You can move and resize the window to meet your needs, and modify it’s contents with right-clicks.
